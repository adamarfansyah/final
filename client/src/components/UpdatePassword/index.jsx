import PropTypes from 'prop-types';
import { useForm } from 'react-hook-form';
import { FormattedMessage } from 'react-intl';

import Input from '@components/Input';
import Button from '@components/Button';
import classes from './style.module.scss';

const UpdatePassword = ({ onSubmit }) => {
  const {
    register,
    handleSubmit,
    formState: { errors },
    getValues,
  } = useForm();

  return (
    <form className={classes.form} onSubmit={handleSubmit(onSubmit)}>
      <div>
        <Input
          type="password"
          name="password"
          label="New Password"
          errors={errors}
          validationSchema={{
            required: 'New Password is required',
            minLength: { value: 6, message: 'Password must be at least 6 characters' },
          }}
          register={register}
          placeholder="Password"
          required
        />
      </div>
      <div>
        <Input
          type="password"
          name="confirmPassword"
          label="Confirm Password"
          errors={errors}
          validationSchema={{
            required: 'Confirm Password is required',
            validate: (value) => value === getValues('password') || 'Passwords do not match',
          }}
          register={register}
          placeholder="Confirm Password"
          required
        />
      </div>
      <div className={classes.btnContainer}>
        <Button className={classes.btn} variant="primary">
          <FormattedMessage id="app_submit" />
        </Button>
      </div>
    </form>
  );
};

UpdatePassword.propTypes = {
  onSubmit: PropTypes.func,
};

export default UpdatePassword;
