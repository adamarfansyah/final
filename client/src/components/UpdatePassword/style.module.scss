@import '@styles/base';

.form {
  width: 100%;
}
