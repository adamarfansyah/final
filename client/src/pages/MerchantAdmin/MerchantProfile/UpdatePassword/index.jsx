import PropTypes from 'prop-types';
import { useForm } from 'react-hook-form';
import { FormattedMessage } from 'react-intl';
import Button from '@components/Button';
import Input from '@components/Input';

const UpdatePassword = ({ onSubmit }) => {
  const {
    register,
    handleSubmit,
    formState: { errors },
    getValues,
  } = useForm();

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <div>
        <Input
          type="password"
          name="password"
          label="Password"
          errors={errors}
          register={register}
          validationSchema={{
            required: 'Password is required',
            minLength: { value: 6, message: 'Password must be at least 6 characters' },
          }}
          placeholder="Password"
          required
        />
      </div>
      <div>
        <Input
          type="password"
          name="confirmPassword"
          label="Confirm Password"
          errors={errors}
          validationSchema={{
            required: 'Confirm Password is required',
            validate: (value) => value === getValues('password') || 'Passwords do not match',
          }}
          register={register}
          placeholder="Password"
          required
        />
      </div>
      <div>
        <Button variant="primary">
          <FormattedMessage id="app_register_title" />
        </Button>
      </div>
    </form>
  );
};

UpdatePassword.propTypes = {
  onSubmit: PropTypes.func,
};

export default UpdatePassword;
