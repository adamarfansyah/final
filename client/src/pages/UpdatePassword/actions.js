import {
  UPDATE_FORGOT_PASSWORD_MERCHANT,
  UPDATE_FORGOT_PASSWORD_MERCHANT_SUCCESS,
  UPDATE_FORGOT_PASSWORD_USER,
  UPDATE_FORGOT_PASSWORD_USER_SUCCESS,
} from './constants';

export const updateForgotPasswordUser = (data, token, cbSuccess) => ({
  type: UPDATE_FORGOT_PASSWORD_USER,
  data,
  token,
  cbSuccess,
});

export const updateForgotPasswordUserSuccess = (data) => ({
  type: UPDATE_FORGOT_PASSWORD_USER_SUCCESS,
  data,
});

export const updateForgotPasswordMerchant = (data, token, cbSuccess) => ({
  type: UPDATE_FORGOT_PASSWORD_MERCHANT,
  data,
  token,
  cbSuccess,
});

export const updateForgotPasswordMerchantSuccess = (data) => ({
  type: UPDATE_FORGOT_PASSWORD_MERCHANT_SUCCESS,
  data,
});
