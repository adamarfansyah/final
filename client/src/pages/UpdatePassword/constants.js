export const UPDATE_FORGOT_PASSWORD_USER = 'UPDATE_FORGOT_PASSWORD_USER';
export const UPDATE_FORGOT_PASSWORD_USER_SUCCESS = 'UPDATE_FORGOT_PASSWORD_USER_SUCCESS';

export const UPDATE_FORGOT_PASSWORD_MERCHANT = 'UPDATE_FORGOT_PASSWORD_MERCHANT';
export const UPDATE_FORGOT_PASSWORD_MERCHANT_SUCCESS = 'UPDATE_FORGOT_PASSWORD_MERCHANT_SUCCESS';
