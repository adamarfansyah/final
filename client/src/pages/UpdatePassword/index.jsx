import { useDispatch } from 'react-redux';
import { useNavigate, useParams } from 'react-router-dom';
import { useEffect } from 'react';
import { encryptData } from '@utils/encrypt';

import UpdatePassword from '@components/UpdatePassword';
import { updateForgotPasswordMerchant, updateForgotPasswordUser } from './actions';
import classes from './style.module.scss';

const UpdatePasswordPage = () => {
  const { role, token } = useParams();
  const dispatch = useDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    if (role !== 'user' && role !== 'merchant') {
      navigate('/');
    }
  }, []);

  const onSubmitUser = (data) => {
    const encryptedPassword = encryptData(data.password);
    const encryptedConfirmPassword = encryptData(data.confirmPassword);
    dispatch(
      updateForgotPasswordUser(
        { password: encryptedPassword, confirmPassword: encryptedConfirmPassword },
        token,
        () => {
          navigate('/auth');
        }
      )
    );
  };

  const onSubmitMerchant = (data) => {
    const encryptedPassword = encryptData(data.password);
    const encryptedConfirmPassword = encryptData(data.confirmPassword);
    dispatch(
      updateForgotPasswordMerchant(
        { password: encryptedPassword, confirmPassword: encryptedConfirmPassword },
        token,
        () => {
          navigate('/auth/merchant');
        }
      )
    );
  };

  if (role === 'user') {
    return (
      <div className={classes.forgotPassword}>
        <img className={classes.image} src="/error.svg" alt="test" />
        <UpdatePassword onSubmit={(data) => onSubmitUser(data)} forgotPassword />
      </div>
    );
  }

  if (role === 'merchant') {
    return (
      <div className={classes.forgotPassword}>
        <img className={classes.image} src="/error.svg" alt="test" />
        <UpdatePassword onSubmit={(data) => onSubmitMerchant(data)} forgotPassword />
      </div>
    );
  }

  return <h1>something wrong</h1>;
};

export default UpdatePasswordPage;
