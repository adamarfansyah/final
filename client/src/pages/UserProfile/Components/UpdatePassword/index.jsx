import { useForm } from 'react-hook-form';
import { useDispatch } from 'react-redux';

import { updateUserPassword } from '@pages/UserProfile/actions';

import { encryptData } from '@utils/encrypt';
import Input from '@components/Input';
import Button from '@components/Button';
import classes from './style.module.scss';

const UpdatePassword = () => {
  const dispatch = useDispatch();
  const {
    register,
    handleSubmit,
    formState: { errors },
    getValues,
  } = useForm();

  const onSubmit = (data) => {
    const encryptedPassword = encryptData(data.password);
    const encryptedConfirmPassword = encryptData(data.confirmPassword);

    dispatch(updateUserPassword({ password: encryptedPassword, confirmPassword: encryptedConfirmPassword }));
  };

  return (
    <div>
      <form className={classes.form} onSubmit={handleSubmit(onSubmit)}>
        <div>
          <Input
            type="password"
            name="password"
            label="New Password"
            errors={errors}
            validationSchema={{
              required: 'New Password is required',
              minLength: { value: 6, message: 'Password must be at least 6 characters' },
            }}
            register={register}
            placeholder="Password"
            required
          />
        </div>
        <div>
          <Input
            type="password"
            name="confirmPassword"
            label="Confirm Password"
            errors={errors}
            validationSchema={{
              required: 'Confirm Password is required',
              validate: (value) => value === getValues('password') || 'Passwords do not match',
            }}
            register={register}
            placeholder="Confirm Password"
            required
          />
        </div>
        <div className={classes.btnContainer}>
          <Button className={classes.btn} variant="primary">
            Submit
          </Button>
        </div>
      </form>
    </div>
  );
};

export default UpdatePassword;
