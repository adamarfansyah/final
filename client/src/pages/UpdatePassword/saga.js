import { call, put, takeLatest } from 'redux-saga/effects';

import { updateForgotPasswordMerchantApi, updateForgotPasswordUserApi } from '@domain/api';

import { setLoading, showPopup } from '@containers/App/actions';
import { UPDATE_FORGOT_PASSWORD_MERCHANT, UPDATE_FORGOT_PASSWORD_USER } from './constants';

function* doUpdateForgotPasswordUser({ data, token, cbSuccess }) {
  yield put(setLoading(true));
  try {
    yield call(updateForgotPasswordUserApi, data, token);
    yield put(showPopup('Success Change Password!', 'Please login with your new password', true));
    cbSuccess && cbSuccess();
  } catch (error) {
    yield put(showPopup('Sorry :(', error.response.data.status));
  }
  yield put(setLoading(false));
}

function* doUpdateForgotPasswordMerchant({ data, token, cbSuccess }) {
  yield put(setLoading(true));
  try {
    yield call(updateForgotPasswordMerchantApi, data, token);
    yield put(showPopup('Success Change Password!', 'Please login with your new password', true));
    cbSuccess && cbSuccess();
  } catch (error) {
    yield put(showPopup('Sorry :(', error.response.data.status));
  }
  yield put(setLoading(false));
}

export default function* updatePasswordSaga() {
  yield takeLatest(UPDATE_FORGOT_PASSWORD_USER, doUpdateForgotPasswordUser);
  yield takeLatest(UPDATE_FORGOT_PASSWORD_MERCHANT, doUpdateForgotPasswordMerchant);
}
