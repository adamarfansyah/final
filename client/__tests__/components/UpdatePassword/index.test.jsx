import { render } from '@utils/testHelper';
import UpdatePassword from '@components/UpdatePassword';

const mockOnSubmit = jest.fn();

let wrapper;

beforeEach(() => {
  wrapper = render(<UpdatePassword onSubmit={mockOnSubmit} />);
});

describe('Component Update Password', () => {
  test('renders form with correct inputs and button', () => {
    const { getByTestId, getByText } = wrapper;
    expect(getByText(/new password/i)).toBeInTheDocument();
    expect(getByText(/confirm password/i)).toBeInTheDocument();
    expect(getByTestId('btn')).toBeInTheDocument();
  });

  test('Should match with snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });
});
