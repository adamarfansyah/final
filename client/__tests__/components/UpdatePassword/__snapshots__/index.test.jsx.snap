// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Update Password Should match with snapshot 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <div>
      <div
        class="input"
        data-testid="input-container"
      >
        <div
          class="label"
          data-testid="input-label"
        >
          Your Password
          <span>
            *
          </span>
        </div>
        <div
          class="contentInput"
        >
          <input
            data-testid="input"
            id="oldPassword"
            name="oldPassword"
            placeholder="Your Password"
            type="password"
            value=""
          />
        </div>
      </div>
    </div>
    <div>
      <div
        class="input"
        data-testid="input-container"
      >
        <div
          class="label"
          data-testid="input-label"
        >
          New Password
          <span>
            *
          </span>
        </div>
        <div
          class="contentInput"
        >
          <input
            data-testid="input"
            id="password"
            name="password"
            placeholder="Password"
            type="password"
            value=""
          />
          <div
            data-testid="visibility-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="VisibilityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="input"
        data-testid="input-container"
      >
        <div
          class="label"
          data-testid="input-label"
        >
          Confirm Password
          <span>
            *
          </span>
        </div>
        <div
          class="contentInput"
        >
          <input
            data-testid="input"
            id="confirmPassword"
            name="confirmPassword"
            placeholder="Confirm Password"
            type="password"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="btnContainer"
    >
      <button
        class="btn btn"
        data-testid="btn"
        type="submit"
      >
        Kirim
      </button>
    </div>
  </form>
</DocumentFragment>
`;
